### DISCUSSION TITLE
Meeting minutes for 28.09.2023

### DISCUSSION THREAD

**Discussion by konradweiss** [Fraunhofer-AISEC/cpg#1319] [Dev Meetings]
### Agenda

1. Open PRs
2. Nodes up for rework and merging.
3. Your wishes for last breaking changes for 8.0
4. Picking up old discussions: Modelling language specific collections(Signature parsing), C++ header dedup

### Discussed PRs

- [x] #1315 
- [x] #1282 
- [x] #1171

**Comment by KuechA** [comment]
# Discussed PRs

## Concerning #1171:
* WIP: dokka plugin to get the JavaDoc for the nodes, edges and fields (@konradweiss)
* Wish: Machine-readable format. Can be easily integrated

## Concerning #1282 
* Limited progress, no real blocking issues (besides the time)

## Concerning #1215 
* Limited progress, no real blocking issues (besides the time)

## Concerning #1315 
* `MagicFixItPass` is questionable: Could it be a cpp-specific pass? => Yes, better naming
* Generalization: interprocedural issue cannot be included with the current approach
* Current approach seems ok for a first draft, "dependence tree" as future work
* Finalize the PR

# Changes for 8.0
* Separate discussion
* Timeframe: Release in 2-3 weeks
* ASAP wishes for node renaming and similar breaking changes

# Wishes for new features/current problems

* Interprocedural data flows with pointers => Slightly hacky solution possible
* Alias or points-to analysis => Difficult, cannot be precisely solved generally
    - Save information "this is an alias" (maybe in DFG or something else)
    - Aliasing in fixed-point information as todo for the future?
* Modelling standard-library functions is easy but simply "has to be implemented"
